### DISCUSSION TITLE
ERROR - (NP.df_utils.auto_normalization_setting) - encountered variable with one unique value

### DISCUSSION THREAD

**Discussion by jmsimon68** [ourownstory/neural_prophet#694] [Q&A - get help using NeuralProphet]
I am trying to compare forecast with and without AR, and I can't make it run with AR.

The model works fine, with future regressors added. But when I try to run it with AR, I get that error, using the same dataframe.

The code I am using is quite basic:

        m = NeuralProphet()
        m.add_future_regressor('var A')
        m.add_future_regressor('var B')
        m.fit(df, freq='M')
        future = m.make_future_dataframe(df, periods=6, regressors_df=df_fut)
        fcst = m.predict(futureX)

This one runs fine. But this one below:

        m2 = NeuralProphet(n_lags=12, n_forecasts=6, weekly_seasonality=False, daily_seasonality=False)
        m2 = m2.add_lagged_regressor('var A')
        m2 = m2.add_lagged_regressor('var B')
        m2.fit(df, freq='M')
        future2 = m2.make_future_dataframe(df, periods=6)
        fcst_2 = m2.predict(future2)

Stops at fit, with the error in the title.

The df is a relatively simple one, with two columns, ds and y, and 50 rows. In the first case, I do provide 'var A' and 'var B' values for the next 6 months.

Any idea what is going on?

THANKS!

**Comment by ourownstory** [comment]
It seems like one of your regressors may have a single value (not changing over time)?

If you provide a minimal code snippet (including some dummy data), I can try to reproduce the error and investigate.

**Reply by jmsimon68** [reply]
The regressors have zero value for the first 27 rows, but it has different values for the last 23 rows. Not sure if that is the problem...

The code is exactly as above. See the data attached.

[data_sample.csv](https://github.com/ourownstory/neural_prophet/files/9423194/data_sample.csv)

Thanks!

**Reply by ourownstory** [reply]
do you maybe do a data-split, where, after accounting for the tabularization (of lags and forecasts), there are only zeros in the training inputs?

**Reply by jmsimon68** [reply]
Hi,

No, I don't do any data-splits after that. This is the code and the two files I am using. As I said, the first part of the code (the one without AR) runs ok, but the second part gives me an error.

`import pandas as pd
from neuralprophet import NeuralProphet

df = pd.read_csv('data_sample.csv')
df_fut = pd.read_csv('fut_data.csv')

m = NeuralProphet()
m.add_future_regressor('var A')
m.add_future_regressor('var B')
m.fit(df, freq='M')
future = m.make_future_dataframe(df, periods=6, regressors_df=df_fut)
fcst = m.predict(future)
print(fcst)

m2 = NeuralProphet(n_lags=12, n_forecasts=6, weekly_seasonality=False, daily_seasonality=False)
m2 = m2.add_lagged_regressor('var A')
m2 = m2.add_lagged_regressor('var B')
m2.fit(df, freq='M')
future2 = m2.make_future_dataframe(df, periods=6)
fcst_2 = m2.predict(future2)
print(fcst_2)`

I am also attaching the 2 CSVs being used.

Thanks for your help!!
[fut_data.csv](https://github.com/ourownstory/neural_prophet/files/9491299/fut_data.csv)
[data_sample.csv](https://github.com/ourownstory/neural_prophet/files/9491300/data_sample.csv)
